Mathlib/LinearAlgebra/CliffordAlgebra/SpinGroup.lean

```diff
@@ -0,0 +1,474 @@
+/-
+Copyright (c) 2022 Jiale Miao. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jiale Miao, Utensil Song, Eric Wieser
+-/
+import Mathlib.GroupTheory.GroupAction.ConjAct
+import Mathlib.Algebra.Star.Unitary
+import Mathlib.LinearAlgebra.CliffordAlgebra.Star
+import Mathlib.LinearAlgebra.CliffordAlgebra.Even
+import Mathlib.LinearAlgebra.CliffordAlgebra.Inversion
+
+/-!
+# The Pin group and the Spin group
+
+In this file we define `lipschitzGroup`, `pinGroup` and `spinGroup` and show they form a group.
+
+## Main definitions
+
+* `lipschitzGroup`: the Lipschitz group with a quadratic form.
+* `pinGroup`: the Pin group defined as the infimum of `lipschitzGroup` and `unitary`.
+* `spinGroup`: the Spin group defined as the infimum of `pinGroup` and `CliffordAlgebra.even`.
+
+## Implementation Notes
+
+Here are some discussion about the latent ambiguity of definition :
+https://mathoverflow.net/q/427881/172242 and https://mathoverflow.net/q/251288/172242
+
+The definition of the Lipschitz group `{𝑥 ∈ 𝐶𝑙(𝑉,𝑞) │ 𝑥 𝑖𝑠 𝑖𝑛𝑣𝑒𝑟𝑡𝑖𝑏𝑙𝑒 𝑎𝑛𝑑 𝑥𝑣𝑥⁻¹∈ 𝑉}` is given by:
+• Fulton, W. and Harris, J., 2004. Representation theory. New York: Springer, p.chapter 20.
+• https://en.wikipedia.org/wiki/Clifford_algebra#Lipschitz_group
+But they presumably form a group only in finite dimensions. So we define `lipschitzGroup` with
+closure of all the invertible elements in the form of `ι Q m`, and we show this definition is
+at least as large as the other definition (See `lipschitzGroup.mem_conjAct_le` and
+`lipschitzGroup.mem_involute_le`). The reverse statement presumably being true only in finite
+dimensions.
+
+## TODO
+
+Try to show the reverse statement is true in finite dimensions.
+-/
+
+variable {R : Type*} [CommRing R]
+
+variable {M : Type*} [AddCommGroup M] [Module R M]
+
+variable {Q : QuadraticForm R M}
+
+section Pin
+
+open CliffordAlgebra MulAction
+
+open scoped Pointwise
+
+/-- `lipschitzGroup` is the subgroup closure of all the invertible elements in the form of `ι Q m`
+where `ι` is the canonical linear map `M →ₗ[R] CliffordAlgebra Q`. -/
+def lipschitzGroup (Q : QuadraticForm R M) :=
+  Subgroup.closure ((↑) ⁻¹' Set.range (ι Q) : Set (CliffordAlgebra Q)ˣ)
+#align lipschitz lipschitzGroup
+
+namespace lipschitzGroup
+
+/-- If x is in `lipschitzGroup Q`, then `(ι Q).range` is closed under twisted conjugation.
+The reverse statement presumably being true only in finite dimensions.-/
+theorem mem_conjAct_le {x : (CliffordAlgebra Q)ˣ} (hx : x ∈ lipschitzGroup Q)
+    [Invertible (2 : R)] :
+    ConjAct.toConjAct x • LinearMap.range (ι Q) ≤ LinearMap.range (ι Q) := by
+  unfold lipschitzGroup at hx
+  apply Subgroup.closure_induction'' hx
```

Comment:
You should be able to use the `induction` tactic here and elsewhere

Reply:
Checked that proofs are now all changed to use `induction`. Thanks!